velox/functions/sparksql/Arithmetic.h

```diff
@@ -368,6 +368,68 @@ FOLLY_ALWAYS_INLINE static int8_t fromHex(char c) {
 }
 } // namespace detail
 
+template <typename T>
+struct WidthBucketFunction {
+  FOLLY_ALWAYS_INLINE bool call(
+      int64_t& result,
+      double value,
+      double min,
+      double max,
+      int64_t numBucket) {
+    // NULL would be returned if the input arguments don't follow conditions
+    // list belows:
+    // - `numBucket` must be greater than zero and be less than Long.MaxValue.
+    // - `value`, `min`, and `max` cannot be NaN.
+    // - `min` bound cannot equal `max`.
+    // - `min` and `max` must be finite.
+    if (shouldReturnsNull(value, min, max, numBucket)) {
+      return false;
+    }
+
+    result = computeBucketNumberNotNull(value, min, max, numBucket);
+    return true;
+  }
+
+  static FOLLY_ALWAYS_INLINE bool
+  shouldReturnsNull(double value, double min, double max, int64_t numBucket) {
+    return numBucket <= 0 || numBucket == std::numeric_limits<int64_t>::max() ||
+        std::isnan(value) || min == max || !std::isfinite(min) ||
+        !std::isfinite(max);
+  }
+
+  static FOLLY_ALWAYS_INLINE int64_t computeBucketNumberNotNull(
+      double value,
+      double min,
+      double max,
+      int64_t numBucket) {
+    double lower = std::min(min, max);
+    double upper = std::max(min, max);
```

Comment:
These variables do not seem to be needed.

Reply:
removed